stages/test/test_grub2_legacy.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python3
+
+import os.path
+
+import pytest
+
+STAGE_NAME = "org.osbuild.grub2.legacy"
+
+
+# Test that the /etc/default/grub file is created with the correct content
+@pytest.mark.parametrize("test_data,expected_conf", [
```

Comment:
(super nitpick) I wonder if the tests could be shared but totally fine to ignore or do in a followup